### PR TITLE
Use elements local property to generate proper JMS mappings

### PR DESCRIPTION
**Pull request by Toilal**
Related to goetas-webservices/soap-client#46

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -51,6 +51,11 @@
       "dev-master": "0.2-dev"
     }
   },
+  "scripts": {
+    "tests": [
+      "php ./vendor/phpunit/phpunit/phpunit -v"
```
Comment (goetas) [inline composer.json]: i think that this is not needed. you can do just `vendor/bin/phpunit` (is short and works the same)

(this is just a minor note, no need to fix it for now)

**Review by goetas — APPROVED** [review]

**Comment by Toilal** [comment]
I'm working right now to narrow down the new test.

**Comment by goetas** [comment]
:+1:

**Comment by goetas** [comment]
The part with the xsd reader has been merged, now I think that should be easier to work on this

**Comment by Toilal** [comment]
Let's try with @dev on xsd-reader.

**Comment by Toilal** [comment]
`COMPOSER_FLAGS='--prefer-lowest --prefer-stable'` build is failing because it doesn't grab the `@dev` version of xsd-reader, but other builds are OK.

**Comment by Toilal** [comment]
Maybe you could perform a release of xsd-reader, so we can merge this pull request with a stable version constraint on this dependency ? Then xsd2php could be released too ? PR on soap-client is only related to tests, so it can wait.

**Comment by goetas** [comment]
https://github.com/goetas-webservices/xsd-reader/releases/tag/v0.3.7 done

**Comment by Toilal** [comment]
Thanks for the release of `xsd-reader`. Now, this PR should be merged and `xsd2php` released too.

File: tests/Converter/JMS/Xsd2JmsGroupTest.php [hunk 2]
```diff
@@ -582,9 +582,9 @@ public function testGeneralParts()
                             'expose' => true,
                             'access_type' => 'public_method',
                             'serialized_name' => 'element-1',
-//                            'xml_element' => array(
-//                                'namespace' => 'http://www.example.com'
-//                            ),
+                            'xml_element' => array(
```
Comment (goetas) [inline tests/Converter/JMS/Xsd2JmsGroupTest.php:585]: is this the whole test? do we need something better?

Comment (Toilal) [reply tests/Converter/JMS/Xsd2JmsGroupTest.php:585]: Sorry, I don't understand your question. What's the problem here ?

Comment (Toilal) [reply tests/Converter/JMS/Xsd2JmsGroupTest.php:585]: Ho I see. You mean the test changes are too short for the fix ? I'll try to add more checks.

Comment (Toilal) [reply tests/Converter/JMS/Xsd2JmsGroupTest.php:585]: In fact, the fix mostly relies on `xsd-reader` changes. `xsd-reader` carries a whole new test class for the use case, so i'm not sure it's required to bring more checks in `xsd2php` module.

**Comment by goetas** [comment]
I'm good with it! Thanks a lot for the hard work!
I have to admit that this libraries needed some maintenance, and you did it! Thanks again

**Pull request merged**